### ISSUE TITLE
Lecture 6 (logistic regression) comments

### ISSUE THREAD

**Issue by psnairne** [P2GX/bio-ontollm#5]
Hi @pnrobinson @leokim-l,

Leo is technically responsible for giving feedback on this lecture according to the [Contribution List](https://charitede.sharepoint.com/:x:/r/sites/AGRobinsonTeam/_layouts/15/Doc.aspx?action=edit&sourcedoc=%7Befac78d0-474a-413e-8a8c-a78c69bc7959%7D&wdExp=TEAMS-TREATMENT&web=1), but I needed to read it in order to understand the background for the lecture afterward on Neural Nets. Lots of comments/typos came to mind as I was reading so it made sense for me to note them down anyway...

I'll try separate the comments into their different categories. 

_General comments:_

1. This lecture assumes that the reader understands the general framework of machine learning (i.e. training, input data, prediction). It essentially assumes that the reader knows linear regression. But the prerequisites in the course requirements are: "Some knowledge of probability, calculus, and algorithms (We will explain everything as we go along)." So IMO either the course requirements should be adjusted, or there needs to be some background on how ML works. 
2. Related to the above comment, I think there should be an extra slide at the start with an "overall" view of linear regression. I.e. there is no slide saying "we have n feature vectors which we are going to use to train our model". 
3. Certain slides are still written with a neural net in mind (e.g. using matrices instead of vectors): Slide 17, Slide 30

_Suggestions_

1. Slide 13: We could show the other plot for the y=0 case too
2. Slide 14: I'm not sure where log p(y|x) has come from. Why don't we just start with y log y hat + (1 - y) log( 1 - y hat ) as the definition for the loss function? 
3. I think it's a tragedy to mention the gradient and not have a 3d diagram showing how it is the direction of steepest increase. Something like: 
![](https://www.geogebra.org/resource/zsnwNmZB/xSbFyDCpppyKxh0y/material-zsnwNmZB.png)
4. Slide 21: I was scared of the chain rule until I watched 3 Blue 1 Brown's video about it - now I love it. I think it would be great if we could point students towards it https://www.youtube.com/watch?v=YG15m2VwSjA
5. Slide 30: I wonder if a two dimensional example f(x_1, x_2) here would be more explanatory for the gradient? Otherwise we are just doing the derivative
6. Slide 38: I think you say that softmax generalises sigmoid for K = 2
7. Slide 38: Should softmax not be kept for the neural net lecture?
8. Slides 47 - 54: If at some point it is decided that this lecture is too long, I thought these details on L2 regularization could be possibly moved to the homework sheet, since I think that is where it will be used primarily

_Typos/minor corrections_

1. Slide 2: "serves as a good preparation" -> "is a good preparation"
2. Slide 5: "8x8 image" -> "8x8 black and white image"
3. Slide 6: " vector of a vector of weights" -> "vector of weights"
4. Slide 6: If I am not mistaken it should be w_j not w_i? 
5. Slide 6: It should say somewhere that we have training datapoints x_1, ..., x_i, ..., x_m
6. Slide 6: Even better, maybe we should just forget the index for the datapoints? In the other slides, you typically assume we have a single datapoint x, and that would simplify things here.
7. Slide 7: Bold notation should be kept for vectors in equation y = w dot x + b
8. Slide 11: the prediction of the class label should be "y" not "y hat". sigma(y) is y hat (the middle step)
9. Slide 13: y_i -> y
10. Slide 13: we need some brackets so it is - log(1 - sigma(z)) I believe
11. Slide 16: previously we have used underscores for y_i, y_i'. Should mention we have m datapoints again.
12. Slide 16: We use the terms objective function and cost function. Maybe it should just be called cost function consistently?
13. Slide 18. It should be stated that f is a function from n dimensional space to R. 
14. Slide 20: firstderivative -> "first derivative"
15. Slide 20: I think we should drop the i index here, it isn't used almost anywhere else, and it confuses the equation
16. Slide 20: I think we should have a reminder of the equation for the cost function L here as well
17. Slide 23: "Using y hat = sigma(wx + b)". I don't think this is necessary, the cost function L seems to typically be presented in terms of y hat
18. Slide 25: "substituing" -> "substituting"
19. Slide 25: Again we should remove the index i here
20. Slide 29: Not sure why it says "derivative of cost". I also don't think it's clear what the gradient arrow is referring to. As I picture in my head, the gradient vectors are running along the X-axis, as a horizontal arrow <------
21. Slide 33: "The noisiness may even be an advantage because it may help avoid falling into local minima". For logistic regression there are not local minima
22. Slide 34: The cost function seems to be alternating between J and C. Maybe just J?
23. Slide 36: "parenters" -> "parameters"
24. Slide 41: "g" -> "y_hat" for consistency
25. Slide 43: "alpha" -> "eta"
26. Slide 54: "logisitic" -> "logistic"
